### ISSUE TITLE
[prefer-string-starts-ends-with] False positive for substring with length limit

### ISSUE THREAD

**Issue by tkalliom** [typescript-eslint/typescript-eslint#4181]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-string-starts-ends-with": ["error"]
  }
}
```

```TS
export function isThreeCharPrefixMatch(str: string, candidate: string): boolean {
  return str.substring(0, 3) === candidate;
}
```

`tsconfig`: the default as generated by `tsc --init`
**Expected Result**
No error, since a `substring` call other than `substring(0)` can not be replaced with `startsWith` alone.

**Actual Result**
The following error was printed:
```
2:10  error  Use 'String#startsWith' method instead  @typescript-eslint/prefer-string-starts-ends-with
```

**Additional Info**
Consider the code with the change suggested by the linter:
```TS
export function isThreeCharPrefixMatchB(str: string, candidate: string): boolean {
  return str.startsWith(candidate);
}
```
Now
```
> isThreeCharPrefixMatch("int32", "in");
false
> isThreeCharPrefixMatchB("int32", "in");
true
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.4.0` |
| `@typescript-eslint/parser`        | `5.4.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `8.2.0` |
| `node`                             | `12.22.5` |
